src/Model/GlobalContact.php

```diff
@@ -0,0 +1,1000 @@
+							GROUP BY `gcontact`.`nurl`
+							ORDER BY `gcontact`.`nurl` DESC
+							LIMIT 1000",
+				intval(local_user()), dbesc(CONTACT_IS_SHARING), dbesc(CONTACT_IS_FRIEND),
+				dbesc(NETWORK_DFRN), dbesc($ostatus), dbesc($diaspora),
+				dbesc(escape_tags($search)), dbesc(escape_tags($search)), dbesc(escape_tags($search))
+			);
+
+			return $results;
+		}
+	}
+
+	/**
+	 * @brief Sanitize the given gcontact data
+	 *
+	 * @param array $gcontact array with gcontact data
+	 * @throw Exception
+	 *
+	 * Generation:
+	 *  0: No definition
+	 *  1: Profiles on this server
+	 *  2: Contacts of profiles on this server
+	 *  3: Contacts of contacts of profiles on this server
+	 *  4: ...
+	 *
+	 */
+	public static function sanitize($gcontact)
+	{
+		if ($gcontact['url'] == "") {
+			throw new Exception('URL is empty');
+		}
+	
+		$urlparts = parse_url($gcontact['url']);
+		if (!isset($urlparts["scheme"])) {
+			throw new Exception("This (".$gcontact['url'].") doesn't seem to be an url.");
+		}
+	
+		if (in_array($urlparts["host"],
+			array("www.facebook.com", "facebook.com", "twitter.com", "identi.ca", "alpha.app.net"))
+		) {
+			throw new Exception('Contact from a non federated network ignored. ('.$gcontact['url'].')');
+		}
+	
+		// Don't store the statusnet connector as network
+		// We can't simply set this to NETWORK_OSTATUS since the connector could have fetched posts from friendica as well
+		if ($gcontact['network'] == NETWORK_STATUSNET) {
+			$gcontact['network'] = "";
+		}
+	
+		// Assure that there are no parameter fragments in the profile url
+		if (in_array($gcontact['network'], array(NETWORK_DFRN, NETWORK_DIASPORA, NETWORK_OSTATUS, ""))) {
+			$gcontact['url'] = self::cleanContactUrl($gcontact['url']);
+		}
+	
+		$alternate = PortableContact::alternateOStatusUrl($gcontact['url']);
+	
+		// The global contacts should contain the original picture, not the cached one
+		if (($gcontact['generation'] != 1) && stristr(normalise_link($gcontact['photo']), normalise_link(System::baseUrl()."/photo/"))) {
+			$gcontact['photo'] = "";
+		}
+	
+		if (!isset($gcontact['network'])) {
+			$r = q(
+				"SELECT `network` FROM `contact` WHERE `uid` = 0 AND `nurl` = '%s' AND `network` != '' AND `network` != '%s' LIMIT 1",
+				dbesc(normalise_link($gcontact['url'])),
+				dbesc(NETWORK_STATUSNET)
+			);
+			if (DBM::is_result($r)) {
+				$gcontact['network'] = $r[0]["network"];
+			}
+	
+			if (($gcontact['network'] == "") || ($gcontact['network'] == NETWORK_OSTATUS)) {
+				$r = q(
+					"SELECT `network`, `url` FROM `contact` WHERE `uid` = 0 AND `alias` IN ('%s', '%s') AND `network` != '' AND `network` != '%s' LIMIT 1",
+					dbesc($gcontact['url']),
+					dbesc(normalise_link($gcontact['url'])),
+					dbesc(NETWORK_STATUSNET)
+				);
+				if (DBM::is_result($r)) {
+					$gcontact['network'] = $r[0]["network"];
+				}
+			}
+		}
+	
+		$gcontact['server_url'] = '';
+		$gcontact['network'] = '';
```

Comment:
This doesn't look right, since the following code makes use of `isset($gcontact['network'])` which would always be true since it's declared as empty string here no matter its value set on lines 152 and 141.

Reply:
It's still weird that it seems to work on 'develop' though.

Reply:
removing that line